src/Sulu/Bundle/SecurityBundle/User/UserProvider.php

```diff
@@ -127,7 +126,7 @@ public function supportsClass($class): bool
 
     public function upgradePassword(PasswordAuthenticatedUserInterface $user, string $newHashedPassword): void
     {
-        if (!$user instanceof User) { // @phpstan-ignore-line we can not be 100% sure about this but generic forces us to check
```

Comment:
Interesting is we didn't here either not trust the PHPStan generic and added a check already.